tests/components/broadlink/test_device.py

```diff
@@ -20,16 +20,13 @@
 async def test_device_setup(hass):
     """Test a successful setup."""
     device = get_device("Office")
-    mock_api = device.get_mock_api()
-    mock_entry = device.get_mock_entry()
-    mock_entry.add_to_hass(hass)
 
-    with patch("broadlink.gendevice", return_value=mock_api), patch.object(
+    with patch.object(
```

Comment:
these object patches should really patch broadlink setups rather than HA's config entries in another PR

Reply:
@ctalkington Do you know why patching out `hass.config_entries.flow` does not work here?
```python
async def test_setup_discover_already_known_host(hass):
    """Test we do not create flows for known devices rediscovered at startup."""
    device = get_device("Living Room")
    mock_api = device.get_mock_api()
    mock_entry = device.get_mock_entry()
    mock_entry.add_to_hass(hass)

    with patch(
        DEVICE_DISCOVERY, return_value=[mock_api]
    ) as mock_discover, patch.object(
        hass.config_entries.flow, "async_init"
    ) as mock_init:
        assert await async_setup_component(hass, DOMAIN, {}) is True

    assert mock_discover.call_count == 1
    assert mock_init.call_count == 0
```
The tests are passing, but the flow is being created, I am sniffing the packets with Wireshark.